### DISCUSSION TITLE
Async bulk / batch poll

### DISCUSSION THREAD

**Discussion by raphaelauv** [confluentinc/parallel-consumer#58] [Ideas] [answered]
Would it be possible for a **bulk poll** ?

exemple :
```java
parallelConsumer.polls(records -> records.forEach( record ->
                  log.info("Concurrently processing a record: {}", record)
                  )
);
```
thank to this interface
```java
void polls(Consumer<ConsumerRecords<K, V>> usersVoidConsumptionFunction);
```

----------

I currently have a consumer that for each result of **poll** make a single async bulk load ( pipelining operation ) to redis ( with redisson lib )

so my flow is 
```
while(true) :

    msgs = consumer.poll()
    bulk_redis_operation = transform(msgs)
    future = bulk_redis_operation.executeAsync()
    future.await()
    consumer.commitSync()
```
    
Thank you

**Comment by astubbs** [comment, marked as answer]
Yes, absolutely! See issue https://github.com/confluentinc/parallel-consumer/issues/18 and POC https://github.com/confluentinc/parallel-consumer/pull/19

Please vote on the issue :)

**Comment by astubbs** [comment]
Do you have thoughts on how you might like to control the quantity of records in the batch? Would a set max quantity be enough?

**Comment by raphaelauv** [comment]
By default it could use the value of (for a set max) of the current consumer
```java
ConsumerConfig.MAX_POLL_RECORDS_CONFIG 
```
?

**Reply by astubbs** [reply]
Hmm interesting, so your use case you think you'd like to bulk process 500 messages at a time?

**Reply by raphaelauv** [reply]
500 is default value , but my application custom that value
```java
    final int maxPollRecord = parseInt(argsMap.get("maxPollRecord"));
    logger.info("Max poll record=" + maxPollRecord);
    consumerConfig.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecord);
    Consumer<String, T> consumer = new KafkaConsumer<>(consumerConfig);
```
